### ISSUE TITLE
Support for dynamically adding/removing monitors

### ISSUE THREAD

**Issue by Cu3PO42** [end-4/dots-hyprland#551]
**What would you like to be added?**  
I would like to dynamically add and remove monitor-specific windows, such as the bar, when new monitors are connected or removed.

**How will it help**  
I commonly plug and unplug my laptop into various monitors/docking stations/... at work and currently need to restart illogical-impulse to have the bar show up on all monitors.

**Extra info**  
I'm happy to implement this if you'll accept a PR.

**Comment by end-4** [comment]
It sounds like a lot of work and I don't use an external monitor, so feel free to make a PR

**Comment by end-4** [comment]
Note that we might have to turn **everything** into callbacks instead of instances first

**Comment by Cu3PO42** [comment]
Yeah, it's definitely a non-trivial endeavour. I did get started on this yesterday, and while it's not as easy as I had hoped (due to GDK not exposing some state that would be really helpful), it seems doable overall.

**Comment by end-4** [comment]
We've switched to Quickshell, and this is now naturally supported thanks to QML's reactivity
Therefore I consider this as solved
